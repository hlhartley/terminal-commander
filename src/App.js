import React, { Component } from 'react';
import './CSS/main.scss';
import GameContainer from './GameContainer';
import TerminalCard from './TerminalCard';
import AllCommandsModal from './AllCommandsModal';
import KeyModal from './KeyModal';

export default class App extends Component {
  constructor() {
    super();

  this.state = {
    hhTerminalCommands: [],
    score: 0,
    showCommands: false,
    showKeys: false,
  }
}

componentDidMount() {
  fetch("http://memoize-datasets.herokuapp.com/api/v1/hhTerminalCommands")
    .then(results => results.json())
    .then((result) => {
      this.setState({
        hhTerminalCommands: result.hhTerminalCommands
      })
    })
    .catch(err => {
      this.setState({error: err})
    })
  }

  updateScore() {
    this.setState({score: this.state.score + 1})
  }

  resetGame = () => {
    this.setState({score: 0})
  }

  toggleAllCommandsModal() {
    this.setState({showCommands: !this.state.showCommands})
  }

  toggleAllKeysModal() {
    this.setState({showKeys: !this.state.showKeys})
  }

  renderModals() {
    if(this.state.showCommands) {
      return ( 
        <AllCommandsModal 
          terminalCommands = {this.state.hhTerminalCommands} 
          toggleModal = {this.toggleAllCommandsModal.bind(this)}
        />
      )
    }
    if(this.state.showKeys) {
      return <KeyModal toggleModal = {this.toggleAllKeysModal.bind(this)}/>
    }
  }

  render() {
    return (
      <div className='app-container'>
        <GameContainer 
          clickReset = {this.state.clickReset}
          score = {this.state.score}
        />
        <div className='buttons-container'>
          <button className='reset-button' onClick={this.resetGame}>RESET GAME</button>
          <button className='keys-button' onClick={this.toggleAllKeysModal.bind(this)}>KEY</button>
          <button className='allcommands-button' onClick={this.toggleAllCommandsModal.bind(this)}>ALL COMMANDS</button>
        </div>
        <TerminalCard 
          terminalCommands = {this.state.hhTerminalCommands}
          score = {this.state.score}
          updateScore = {this.updateScore.bind(this)}
        />
        {this.renderModals()}
      </div>
    );
  }
}