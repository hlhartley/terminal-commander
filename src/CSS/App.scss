.app-container {
  display: grid;
  .buttons-container {
    display: flex;
    justify-content: center;
    background-color: #00522e;
  }
  .reset-button, .keys-button, .allcommands-button {
    // position: absolute;
    margin: 10px;
    width: 130px;
    border-radius: 15px;
    top: 15px;
    right: 18px;
    font-size: 15px;
    padding: 3px;
    color: #FFF;
    background-color: #5c3e3e;
    border: 2px solid #568556;
    letter-spacing: 1.7px;
    opacity: .8;
    cursor: pointer;
    &:hover {
      background-color: #67d4e7;
      opacity: 1;
    }
  }
    .keys-button {
      top: 49px;
      }
      .allcommands-button {
        top: 83px;
      }
}

