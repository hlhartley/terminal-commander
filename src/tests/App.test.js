import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { shallow } from 'enzyme';

const hhTerminalCommands = [{
  id: 1,
  description: "Reset terminal screen",
  command: "reset"
  },
  {
  id: 2,
  description: "Clear screen",
  command: "clear"
  }]

let wrapper;
describe('App', () => {

  beforeEach(() => {
    wrapper = shallow(
      <App {...hhTerminalCommands}/>
    );
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have a proper default state', () => {
    expect(wrapper.state()).toEqual({
      hhTerminalCommands: [],
      score: 0,
      showCommands: false,
      showKeys: false,
    })
  })
})

// Remaining Tests:
// updateScore
// resetGame
// toggleAllCommandsModal
// toggleAllKeysModal
// renderModals
// buttons?
