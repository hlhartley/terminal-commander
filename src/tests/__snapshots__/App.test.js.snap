// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    0={
      Object {
        "command": "reset",
        "description": "Reset terminal screen",
        "id": 1,
      }
    }
    1={
      Object {
        "command": "clear",
        "description": "Clear screen",
        "id": 2,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <GameContainer
          score={0}
        />,
        <div
          className="buttons-container"
        >
          <button
            className="reset-button"
            onClick={[Function]}
          >
            RESET GAME
          </button>
          <button
            className="keys-button"
            onClick={[Function]}
          >
            KEY
          </button>
          <button
            className="allcommands-button"
            onClick={[Function]}
          >
            ALL COMMANDS
          </button>
        </div>,
        <TerminalCard
          score={0}
          terminalCommands={Array []}
          updateScore={[Function]}
        />,
        undefined,
      ],
      "className": "app-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "clickReset": undefined,
          "score": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="reset-button"
              onClick={[Function]}
            >
              RESET GAME
            </button>,
            <button
              className="keys-button"
              onClick={[Function]}
            >
              KEY
            </button>,
            <button
              className="allcommands-button"
              onClick={[Function]}
            >
              ALL COMMANDS
            </button>,
          ],
          "className": "buttons-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "RESET GAME",
              "className": "reset-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "RESET GAME",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "KEY",
              "className": "keys-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "KEY",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ALL COMMANDS",
              "className": "allcommands-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "ALL COMMANDS",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "score": 0,
          "terminalCommands": Array [],
          "updateScore": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <GameContainer
            score={0}
          />,
          <div
            className="buttons-container"
          >
            <button
              className="reset-button"
              onClick={[Function]}
            >
              RESET GAME
            </button>
            <button
              className="keys-button"
              onClick={[Function]}
            >
              KEY
            </button>
            <button
              className="allcommands-button"
              onClick={[Function]}
            >
              ALL COMMANDS
            </button>
          </div>,
          <TerminalCard
            score={0}
            terminalCommands={Array []}
            updateScore={[Function]}
          />,
          undefined,
        ],
        "className": "app-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "clickReset": undefined,
            "score": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="reset-button"
                onClick={[Function]}
              >
                RESET GAME
              </button>,
              <button
                className="keys-button"
                onClick={[Function]}
              >
                KEY
              </button>,
              <button
                className="allcommands-button"
                onClick={[Function]}
              >
                ALL COMMANDS
              </button>,
            ],
            "className": "buttons-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "RESET GAME",
                "className": "reset-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "RESET GAME",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "KEY",
                "className": "keys-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "KEY",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ALL COMMANDS",
                "className": "allcommands-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "ALL COMMANDS",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "score": 0,
            "terminalCommands": Array [],
            "updateScore": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
